### ISSUE TITLE
[Bug]: Txn Workspace issue in v1.1.1

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#14597]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

v1.1.1

### Commit ID

9d616832a

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

```
Traceback (most recent call last):
  File "/Users/qinshuqi/tmp/issue-mo-14577/test.py", line 47, in <module>
    run()
  File "/Users/qinshuqi/tmp/issue-mo-14577/test.py", line 42, in run
    session.execute(sql_insert, {"id": i, "data": to_db_binary(arr)})
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2229, in execute
    return self._execute_internal(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2133, in _execute_internal
    result = conn.execute(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1414, in execute
    return meth(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 486, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1638, in _execute_clauseelement
    ret = self._execute_context(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1842, in _execute_context
    return self._exec_single_context(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1983, in _exec_single_context
    self._handle_dbapi_exception(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2326, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1964, in _exec_single_context
    self.dialect.do_execute(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 748, in do_execute
    cursor.execute(statement, parameters)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pymysql/cursors.py", line 158, in execute
    result = self._query(query)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pymysql/cursors.py", line 325, in _query
    conn.query(q)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pymysql/connections.py", line 549, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pymysql/connections.py", line 779, in _read_query_result
    result.read()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pymysql/connections.py", line 1157, in read
    first_packet = self.connection._read_packet()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pymysql/connections.py", line 709, in _read_packet
    raise err.InternalError(
sqlalchemy.exc.InternalError: (pymysql.err.InternalError) Packet sequence number wrong - got 2 expected 1
[SQL: insert into t4 (a, b) values(%(id)s, decode(%(data)s,'hex') );]
[parameters: {'id': 1834, 'data': b'2eab763e6933163f8f26613fff2e093ffa53983e861dcd3e0981693db768373fa078e83ec634363f28a4553ee883b43e56a3453f5a51563e74ce173f6c515e3f29b4bc3dcaa8183fddda ... (727 characters truncated) ... fc73d82427c3ff7dd283f34672b3ff0a3443f34020e3e75e8f33e052ee43ca4d39a3e018b0c3f09bb3b3fd6267c3fa83a093e99d3153f50270e3f48d07b3f2221373f96ac3e3f21509c3d'}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
Exception during reset or similar
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 991, in _finalize_fairy
    fairy._reset(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1440, in _reset
    pool._dialect.do_rollback(self)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 657, in do_rollback
    dbapi_connection.rollback()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pymysql/connections.py", line 481, in rollback
    self._execute_command(COMMAND.COM_QUERY, "ROLLBACK")
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pymysql/connections.py", line 797, in _execute_command
    raise err.InterfaceError(0, "")
pymysql.err.InterfaceError: (0, '')

```

### Expected Behavior

no error

### Steps to Reproduce
1. git clone mo
2. git checkout v1.1.1
3. start mo
4. run the following script(from https://github.com/matrixorigin/matrixone/issues/14577#issue-2139394542):
```python
"""
CREATE DATABASE a;
USE a;
CREATE TABLE t4 (a int, b vecf32(128));
"""

import binascii
import time

import numpy as np
from sqlalchemy import create_engine, text
from sqlalchemy.orm import sessionmaker

table_name = "t4"
vec_len = 128
num_inserts = 100_000
num_vector_per_insert = 1


def to_db_binary(value, dim=None):
    if value is None:
        return value

    value = np.asarray(value, dtype='<f')

    if value.ndim != 1:
        raise ValueError('expected ndim to be 1')

    # return value.tobytes()
    return binascii.b2a_hex(value)


def run():
    engine = create_engine("mysql+pymysql://root:111@127.0.0.1:6001/a")
    Session = sessionmaker(bind=engine)
    session = Session()

    sql_insert = text("insert into t4 (a, b) values(:id, decode(:data,'hex') );")
    for i in range(num_inserts * num_vector_per_insert):
        arr = np.random.rand(vec_len)
        arr = np.asarray(arr, dtype='<f')
        session.execute(sql_insert, {"id": i, "data": to_db_binary(arr)})
    session.commit()


start = time.time()
run()
duration = time.time() - start
print(f"Result: vector dim={vec_len} vectors "
      f"inserted={num_inserts * num_vector_per_insert} "
      f"insert/second={num_inserts * num_vector_per_insert / duration}")
```

**Comment by arjunsk** [comment]
This is the error in execution.
```
/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/bin/python /Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/fvec_sql_insert.py 
Traceback (most recent call last):
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1969, in _exec_single_context
    self.dialect.do_execute(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 922, in do_execute
    cursor.execute(statement, parameters)
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 179, in execute
    res = self._query(mogrified_query)
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 330, in _query
    db.query(q)
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/MySQLdb/connections.py", line 257, in query
    _mysql.connection.query(self, query)
MySQLdb.OperationalError: (20101, 'internal error: panic runtime error: slice bounds out of range [1833:1]: \nruntime.goPanicSliceBU\n\t/usr/local/go/src/runtime/panic.go:160\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*Transaction).adjustUpdateOrderLocked\n\t/Users/arjunsunilkumar/GolandProjects/matrixone-ro/matrixone/pkg/vm/engine/disttae/types.go:330\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*Transaction).Adjust\n\t/Users/arjunsunilkumar/GolandProjects/matrixone-ro/matrixone/pkg/vm/engine/disttae/types.go:317\ngithub.')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/fvec_sql_insert.py", line 47, in <module>
    run()
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/fvec_sql_insert.py", line 42, in run
    session.execute(sql_insert, {"id": i, "data": to_db_binary(arr)})
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 2308, in execute
    return self._execute_internal(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 2199, in _execute_internal
    result = conn.execute(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1416, in execute
    return meth(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 517, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1639, in _execute_clauseelement
    ret = self._execute_context(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1848, in _execute_context
    return self._exec_single_context(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1988, in _exec_single_context
    self._handle_dbapi_exception(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2344, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1969, in _exec_single_context
    self.dialect.do_execute(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 922, in do_execute
    cursor.execute(statement, parameters)
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 179, in execute
    res = self._query(mogrified_query)
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 330, in _query
    db.query(q)
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/MySQLdb/connections.py", line 257, in query
    _mysql.connection.query(self, query)
sqlalchemy.exc.OperationalError: (MySQLdb.OperationalError) (20101, 'internal error: panic runtime error: slice bounds out of range [1833:1]: \nruntime.goPanicSliceBU\n\t/usr/local/go/src/runtime/panic.go:160\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*Transaction).adjustUpdateOrderLocked\n\t/Users/arjunsunilkumar/GolandProjects/matrixone-ro/matrixone/pkg/vm/engine/disttae/types.go:330\ngithub.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*Transaction).Adjust\n\t/Users/arjunsunilkumar/GolandProjects/matrixone-ro/matrixone/pkg/vm/engine/disttae/types.go:317\ngithub.')
[SQL: insert into t4 (a, b) values(%s, decode(%s,'hex') );]
[parameters: (1834, b'7a09993ea209383f258f343e8515a93e7ec9073f35c53f3e27fcd13d7b6a153fe431993e79564f3e6b75143fe1a9f93e266e793fb2438a3e9564643eaad6183f55604d3ecd2efa3ea260 ... (727 characters truncated) ... 2b03ebc62183ec7bc113ea6c9f73e4c8f873d6758843e537b7a3f824a0f3fded4c73e77f66d3e60b0003f88f5573f7ceb073fe6ba653f3836de3e14b64d3f1ef76f3fc772483f498a0c3e')]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
Exception during reset or similar
Traceback (most recent call last):
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 988, in _finalize_fairy
    fairy._reset(
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/pool/base.py", line 1438, in _reset
    pool._dialect.do_rollback(self)
  File "/Users/arjunsunilkumar/PycharmProjects/mo-benchmark-test/venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 692, in do_rollback
    dbapi_connection.rollback()
MySQLdb.OperationalError: (2013, 'Lost connection to MySQL server during query')

Process finished with exit code 1


```

**Comment by nnsgmsone** [comment]
no process

**Comment by nnsgmsone** [comment]
no proccess

**Comment by nnsgmsone** [comment]
fixed by https://github.com/matrixorigin/matrixone/pull/14685

**Comment by nnsgmsone** [comment]
fixed by #14830

**Comment by heni02** [comment]
confirm,closed
commit id:1.1-dev afb4934ae327b7f0b10add4413301524be2221a6
<img width="844" alt="企业微信截图_359e30b8-599c-4516-b793-f4d2813377ca" src="https://github.com/matrixorigin/matrixone/assets/113406637/ae005a47-55b5-415a-b6de-d8ebe1b5b8ee">
